kitchen/modules/child-trie/src/lib.rs

```diff
@@ -1 +1,149 @@
-// child trie storage example
\ No newline at end of file
+/// Child Trie Storage Example (designed to be minimal and generic)
+/// - for efficient storage of (key, value) pairs when **proofs of key inclusion** need to be cheap
+///
+/// - the root of the child trie is used to prove participation of the `key`
+/// - the trie is retained for efficient usage of the associated `value` later
+use support::{
+	decl_module, decl_storage, decl_event, storage::child, ensure,
+	traits::Get, dispatch::Result,
+};
+use system::ensure_signed;
+use substrate_primitives::storage::well_known_keys::CHILD_STORAGE_KEY_PREFIX;
+use primitives::{Blake2Hasher, Hasher};
+
+// TODO: check if this is unnecessary dependency
+use parity_scale_codec::{Encode, Decode};
+use rstd::vec::Vec; // TODO: should consider replacing all `use rstd::prelude::*;` w/ this (consider a gist for this common question)?
```

Comment:
yeah but using prelude is a best practice imo; you won't die from it. But yeah in substrate we tend to never import sth live `Vec` directly from rstd; just pull the prelude

Reply:
Good to know.